src/gpu/intel/sycl/engine.hpp

```diff
@@ -147,6 +123,85 @@ class engine_t : public gpu::intel::compute::compute_engine_t {
         return jitter->get_kernel(*kernel, this);
     }
 
+#ifdef DNNL_EXPERIMENTAL_SELF_SUFFICIENT_L0
+    status_t create_kernels(std::vector<gpu::intel::compute::kernel_t> *kernels,
+            const std::vector<const char *> &kernel_names,
+            const gpu::intel::compute::kernel_ctx_t &kernel_ctx)
+            const override {
+        if (kind() != engine_kind::gpu) {
+            assert(!"not expected");
+            return status::invalid_arguments;
+        }
+        namespace syclex = ::sycl::ext::oneapi::experimental;
+
+        auto device = utils::downcast<const impl::xpu::sycl::engine_impl_t *>(
+                impl())
+                              ->device();
+        VERROR_ENGINE(
+                device.ext_oneapi_can_compile(syclex::source_language::opencl),
+                status::runtime_error,
+                "SYCL OpenCL online compiler not supported on device");
+
+        const char *source = nullptr;
+        for (size_t i = 0; source == nullptr && i < kernel_names.size(); i++)
+            source = ocl::get_kernel_source(kernel_names[i]);
+        VERROR_ENGINE(source, status::runtime_error,
+                "No OpenCL source was found for kernel");
+
+        std::stringstream pp_code;
+        CHECK(gpu::intel::ocl::preprocess_headers(pp_code, source, kernel_ctx));
+
+        std::string build_options = kernel_ctx.options();
+        build_options += " " + device_info()->get_cl_ext_options();
+
+        std::unique_ptr<::sycl::kernel_bundle<::sycl::bundle_state::executable>>
+                kb;
+        auto kb_src = ::sycl::kernel_bundle<
+                ::sycl::bundle_state::ext_oneapi_source>(
+                syclex::create_kernel_bundle_from_source(context(),
+                        syclex::source_language::opencl, pp_code.str()));
```

Comment:
Doesn't it already return the needed type?

```suggestion
        auto kb_src =
                syclex::create_kernel_bundle_from_source(context(),
                        syclex::source_language::opencl, pp_code.str());
```